### PR TITLE
knowledge_plugins.functions.Function: reorganise

### PR DESCRIPTION
**Pull request by Pamplemousse**
By extracting encoding / decoding logic into its own class.

### CODE REVIEW

File: angr/knowledge_plugins/functions/function.py [hunk 1]
```diff
@@ -13,10 +13,10 @@
 from archinfo.arch_arm import get_real_address_if_arm
 import claripy
 
-from ...protos import function_pb2, primitives_pb2
+from .function_parser import FunctionParser
```
Comment (ltfish) [inline angr/knowledge_plugins/functions/function.py]: Import `...` first, `.` last.

**Comment by ltfish** [comment]
CI failed. @Pamplemousse fix CI and I'll hit merge.

**Comment by ltfish** [comment]
Linting is still failing.

**Comment by Pamplemousse** [comment]
Corrected some linting issues that came from mis-distributing `import`s across files in the original commit.
Did some minor cleaning in the `FunctionParser` content, so I added it too.

**Comment by ltfish** [comment]
Merged (with some changes to api-doc waiting to be done). Thanks!

**Pull request merged**